_posts/2023-10-11-specify-file-audit-log-encoding.adoc

```diff
@@ -0,0 +1,34 @@
+---
+layout: post
+title: 'Change the default encoding of the audit log file in WildFly 29'
+date: 2023-10-11
+tags: audit-logging encoding
+synopsis:  An overview of how to change the default encoding of the audit log file.
+author: ivassile
+---
+
+The default encoding used for the audit log file is UTF-8. You can change the default encoding by specifying `encoding` attribute in `file-audit-log`, `periodic-rotating-file-audit-log` or `size-rotating-file-audit-log` elements in the Elytron subsystem. 
+Possible values are: `UTF-8`, `UTF-16BE`, `UTF-16LE`, `UTF-16`, `US-ASCII` or `ISO-8859-1`.
+
+[source,xml]
+----
+<file-audit-log name="local-file" path="audit.log" relative-to="jboss.server.log.dir" synchronized="false" autoflush="true" format="JSON" encoding="UTF-16"/>
+...
+<periodic-rotating-file-audit-log name="periodic-rotating" path="audit.log" relative-to="jboss.server.log.dir" format="JSON" encoding="US-ASCII" suffix="y-M-d"/>
+...
+<size-rotating-file-audit-log name="size-rotating" path="audit.log" relative-to="jboss.server.log.dir" format="JSON" encoding="ISO-8859-1" max-backup-index="5" rotate-on-boot="true" rotate-size="5" suffix="y-M-d"/>
+----
+
+== Example
+
+You can test it by adding the above configuration in the Elytron subsystem and use any of the examples https://github.com/wildfly-security-incubator/elytron-examples/tree/main/[here].
+
+The following CLI command adds a new audit log file with `UTF-16` encoding:
+[source,shell]
+----
+/subsystem=elytron/file-audit-log="local-file-UTF-16":add(path="audit-UTF-16.log", relative-to="jboss.server.log.dir", format="JSON", synchronized="false", encoding="UTF-16")
```

Comment:
Thanks for adding this @ivassile! One final thought - do you think it would be possible to show how to make use of this `file-audit-log` resource? i.e., how to take something like the simple-webapp example, update its security domain configuration to enable audit logging and inspect the resulting file?

Reply:
Updated the post with an example how to make use of the file-audit-log. Updated the date.